### DISCUSSION TITLE
How can I grab the form field data and use in the workflow

### DISCUSSION THREAD

**Discussion by horizontech-ernest** [AOT-Technologies/forms-flow-ai#1239] [Q&A] [answered]
Hi, I am new to the formsflow ai. I am making an approval workflow. I want to split the workflow and assign to different parties to be an approver based on the forms field value user selected.

For example, I have a radio option (which is Programme) in the application form. I want to have different approver to the form based on user selected Programme. If the user selects MBA programme, I want to assign an MBA programme officer to approve/reject the application.

Therefore, what should I do to achieve my purpose? Is there any special setup in the form or workflow? Or I need to create an api to work for it?

**Comment by auslin-aot** [comment, marked as answer]
Use FormBPMDataPipelineListener which copies form data to camunda.

Now form data will be available in the execution.

Write a  script before the approver task to set the assignee based on the value from the form field.
 
![image](https://user-images.githubusercontent.com/99173163/220884622-cc174e55-40ad-48cc-b249-72e4bbe16e78.png)

Then in the approver task use the variable in the assignee/candidate group.
![image](https://user-images.githubusercontent.com/99173163/220887482-c5411c4f-9b36-4f49-a196-9af9c616fa69.png)

Instead of a script, you can also use a dmn to decide on the assignee.

**Reply by horizontechcomhk-fai** [reply]
I have followed the guide to add the class "FormBPMDataPipelineListener" to the sequence flow by using the example form and example two-step bpmn.  However, I still cannot get the variable by key, i.e. "businessOperatingName", from execution. Is there anything I have missed? Thanks.

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/21030708/221410773-d7a6d805-ef82-43a0-921f-e890ac3a1186.png">

<img width="791" alt="image" src="https://user-images.githubusercontent.com/21030708/221410886-5977648c-5bf9-4980-b7f0-e02173247b39.png">

<img width="1722" alt="image" src="https://user-images.githubusercontent.com/21030708/221411059-7bb66ad1-d18c-4b07-bcdc-1e688635834d.png">

**Comment by auslin-aot** [comment]
This may be due to the order of using the Listener & script. The script should have been executed before the listener captured the values.
Can you check by giving the script in the user task?

**Reply by horizontechcomhk-fai** [reply]
THANKS! It works.

When the script is put in a user task, the form variables (key and values) are logged.

We are planning to add one more sequence flow before the first user task in order to get form variables to assign the assignee before reviewing. Hope it works.

<img width="847" alt="image" src="https://user-images.githubusercontent.com/21030708/221576380-5d568451-9f6a-4123-9d39-4c35194e37bd.png">
